### ISSUE TITLE
Fatal error running backup: pruneBackups

### ISSUE THREAD

**Issue by LindezaGrey** [offen/docker-volume-backup#163]
**What are you trying to do?**

i am running my backups and store them in cloudflare's new R2 service.
When the old backups should be pruned, i receive the error below.

It seems to come from this line:

https://github.com/offen/docker-volume-backup/blob/92b888e72c090002a6ff6ecc0a2ce440df7b16ff/internal/storage/s3/s3.go#L116

Would it be in the scope of this tool to handle this error? E.g. to try and skip the metadata on failure?
Strangely this only started after i introduced a folder structure with "AWS_S3_PATH=folder1"

**What is your current configuration?**

offen/docker-volume-backup:v2.21.0

**Log output**
> Fatal error running backup: pruneBackups: error looking up candidates from remote storage: ListObjectsV2 search parameter **metadata** not implemented,

**Addtional Context**
R2 implements a specific S3 feature set. The general documentation can be found here:

https://developers.cloudflare.com/r2/data-access/s3-api/api/

The specs for the ListObjectsV2 method here:

https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html

**Comment by m90** [comment]
I think this is a bug. I cannot really remember why `WithMetaData` is being passed, but I assume it's in order to get the `mtime` of the objects in the bucket. However, if I look at the type definition:

```go
/ ObjectInfo container for object metadata.
type ObjectInfo struct {
	// An ETag is optionally set to md5sum of an object.  In case of multipart objects,
	// ETag is of the form MD5SUM-N where MD5SUM is md5sum of all individual md5sums of
	// each parts concatenated into one string.
	ETag string `json:"etag"`

	Key          string    `json:"name"`         // Name of the object
	LastModified time.Time `json:"lastModified"` // Date and time the object was last modified.
	Size         int64     `json:"size"`         // Size in bytes of the object.
	ContentType  string    `json:"contentType"`  // A standard MIME type describing the format of the object data.
	Expires      time.Time `json:"expires"`      // The date and time at which the object is no longer able to be cached.

	// Collection of additional metadata on the object.
	// eg: x-amz-meta-*, content-encoding etc.
	Metadata http.Header `json:"metadata" xml:"-"`

	// x-amz-meta-* headers stripped "x-amz-meta-" prefix containing the first value.
	UserMetadata StringMap `json:"userMetadata"`

	// x-amz-tagging values in their k/v values.
	UserTags map[string]string `json:"userTags"`

	// x-amz-tagging-count value
	UserTagCount int

	// Owner name.
	Owner Owner

	// ACL grant.
	Grant []Grant

	// The class of storage used to store the object.
	StorageClass string `json:"storageClass"`

	// Versioning related information
	IsLatest       bool
	IsDeleteMarker bool
	VersionID      string `xml:"VersionId"`

	// x-amz-replication-status value is either in one of the following states
	// - COMPLETE
	// - PENDING
	// - FAILED
	// - REPLICA (on the destination)
	ReplicationStatus string `xml:"ReplicationStatus"`

	// Lifecycle expiry-date and ruleID associated with the expiry
	// not to be confused with `Expires` HTTP header.
	Expiration       time.Time
	ExpirationRuleID string

	Restore *RestoreInfo

	// Error
	Err error `json:"-"`
}
```

it doesn't really look as if "metadata" (which seems to be an entirely user defined thing) is needed in order to get `LastModified`.

In case you want to open a PR that fixes this I am happy to merge it and do a release, otherwise I'll see when I find the time to do this. Thanks for the detailed issue btw, this is much appreciated.

**Comment by m90** [comment]
I am pretty sure this option shouldn't be there, it's just been dangling around since the first commit without any reason: https://github.com/offen/docker-volume-backup/pull/19/commits/8c99ec0bdf9c449183e26da5c91586f82ebd195e#diff-9e185f29fa355d7dd8fdd9c9ff1d0723b85206aa7d37c4eec93997005dc291ebR299

**Comment by m90** [comment]
Yeah I just ran a test and a proper `LastModified` value will be returned no matter if `WithMetadata` is being used or not. This means, it can just be removed, which should solve your problem here.

**Comment by m90** [comment]
This should be fixed in `v2.22.1`. Let me know if you run into any further problems when using this image against R2.

**Comment by LindezaGrey** [comment]
Wow this was some fast action! Thanks for the investigation and the new release.
I would have tried to submit a PR but i don`t know Go 🥲 

With the new version all Backups have completed without error.
